### PR TITLE
2779-formulary-parsing-failed(PolicyReporter/requests#2779)

### PR DESCRIPTION
**Pull request by asheeshrana**
2779-formulary-parsing-failed(PolicyReporter/requests#2779)
- Added check to determine horizontal/vertical lines properly

How I tested it:

On the formulary document attached [3653_191374.pdf](https://github.com/PolicyReporter/tabula-java/files/2367247/3653_191374.pdf) executed the following command in vagrant to reproduce the issue:

~~~
java -jar /usr/local/share/tabulapdf/tabula-java-bpi.jar -p all -l -f JSON /opt/pr/policyr/3653_191374.pdf
~~~

After doing the fix... I created and uploaded the newer version of tabula-java rpm in the utils01 server then ran the command below:
~~~
sudo yum clean expire-cache
sudo yum update tabula-java-bpi

java -jar /usr/local/share/tabulapdf/tabula-java-bpi.jar -p all -l -f JSON /opt/pr/policyr/3653_191374.pdf
~~~

This time it ran without any issue

Miscellaneous note:
Basically the fix is as follows:
The logic that checks whether a line is horizontal or vertical allows for slight deviation. The fix here makes an additional check that if a line is perfectly vertical (i.e. x coordinates match) then no need to take slight deviation into account while determining if the line is vertical and same when checking whether the line is horizontal.

---
This change is [Reviewable](https://reviewable.io/reviews/policyreporter/tabula-java/3)

### CODE REVIEW

**Review by chuckeeguo — APPROVED** [review]
> Reviewed 1 of 1 files at r1.
> *[Reviewable](https://reviewable.io/reviews/policyreporter/tabula-java/3)* status: :shipit: complete! all files reviewed, all discussions resolved

*[a discussion](https://reviewable.io/reviews/policyreporter/tabula-java/3#-LM8uXxq3up0IoPnhgUA:-LM8uXxq3up0IoPnhgUB:bfwbmgf) (no related file):*
>Exception in thread "main" java.lang.IllegalArgumentException: lines must be orthogonal, vertical and horizontal
        at technology.tabula.Ruling.intersectionPoint(Ruling.java:216)
        at technology.tabula.Ruling.findIntersections(Ruling.java:385)

confirmed this no longer happens in this branch :lgtm: 

---
